### PR TITLE
OfflineError and OfflineRouteError: to kotlin

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Migrate Java to Kotlin `OfflineError` and `OfflineRouteError` [#issue](https://github.com/mapbox/navigation-sdks/issues/167)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by RingerJK** [comment]
@Guardiola31337 PTAL

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+data class OfflineError(
+    val message: String
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt]: Should we add anything special here as this is a `public` class and can be used from Java?

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.java [hunk 2]
```diff
@@ -55,6 +55,8 @@ public void onFailure_errorIsProvided() {
     Call call = mock(Call.class);
     Throwable throwable = mock(Throwable.class);
 
+    when(throwable.getMessage()).thenReturn("Exception");
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.java]: NIT no need empty lines (L#57 is 👍)

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt (hunk 1 above)
Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt]: nope, `data class` is a class but with overrides `equal` and `hashCode` func

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.java (hunk 2 above)
Comment (RingerJK) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/TarFetchedCallbackTest.java]: done 👍

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt (hunk 1 above)
Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt]: Yeah I know 😅 but now `String` is not-null, wondering if we should add `@NotNull` annotation for Java clients.

Comment (RingerJK) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt]: it's actually NonNull now for Java's users :)

Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineError.kt]: Ohh yeah, that's right the compiler does actually that 👍

**Pull request merged**